at_secondary/at_secondary_server/lib/src/verb/metrics/metrics_impl.dart

```diff
@@ -275,3 +280,100 @@ class LastPkamMetricImpl implements MetricProvider {
     return 'LastPkam';
   }
 }
+
+class NotificationsMetricImpl implements MetricProvider {
+  static final NotificationsMetricImpl _singleton = NotificationsMetricImpl._internal();
+
+  NotificationsMetricImpl._internal();
+  factory NotificationsMetricImpl.getInstance() {
+    return _singleton;
+  }
+  String _asString(dynamic enumData) {
+    return enumData == null ? 'null' : enumData.toString().split('.')[1];
+  }
+
+  @override
+  Future<String?> getMetrics({String? regex}) async {
+    Map<String, dynamic> _metricsMap = <String, dynamic>{
+      "total": 0,
+      "type": <String, int>{
+        "sent": 0,
+        "received": 0,
+      },
+      "status": <String, int>{
+        "delivered": 0,
+        "failed": 0,
+        "queued": 0,
+      },
+      "operations": <String, int>{
+        "update": 0,
+        "delete": 0,
+      },
+      "messageType": <String, int>{
+        "key": 0,
+        "text": 0,
+      },
+      "createdOn": 0,
+    };
+    _metricsMap = await getNotificationStats(_metricsMap);
+    return jsonEncode(_metricsMap);
+  }
+
+  Future<Map<String, dynamic>> getNotificationStats(Map<String, dynamic> _metrics) async {
+    AtNotificationKeystore notificationKeystore = AtNotificationKeystore.getInstance();
+    List notificationsList = await notificationKeystore.getValues();
+    _metrics['total'] = notificationsList.length;
+    for (var notifications in notificationsList) {
+      if (_asString(notifications.toJson()['type']) == 'sent') {
```

Comment:
lot of code duplication instead of 
_asString(notifications.toJson()['type']) == 'sent'

you can have a method like
bool check(type,sent) {
return _asString(notifications.toJson()['type']) == 'sent';
}

Reply:
that works too. I will update it.